pinot-tools/src/main/java/com/linkedin/pinot/tools/admin/command/CreateSegmentCommand.java

```diff
@@ -323,18 +325,16 @@ public boolean accept(File dir, String name) {
       }
       segmentGeneratorConfig.setReaderConfigFile(_readerConfigFile);
     }
-    if (_enableStarTreeIndex) {
-      segmentGeneratorConfig.setEnableStarTreeIndex(true);
-    }
+
     if (_starTreeIndexSpecFile != null) {
-      if (segmentGeneratorConfig.getStarTreeIndexSpecFile() != null
-          && !segmentGeneratorConfig.getStarTreeIndexSpecFile().equals(_starTreeIndexSpecFile)) {
-        LOGGER.warn(
-            "Find starTreeIndexSpecFile conflict in command line and config file, use config in command line: {}",
-            _starTreeIndexSpecFile);
-      }
-      segmentGeneratorConfig.setStarTreeIndexSpecFile(_starTreeIndexSpecFile);
+      StarTreeIndexSpec starTreeIndexSpec = StarTreeIndexSpec.fromFile(new File(_starTreeIndexSpecFile));
+
+      // Specifying star-tree index file enables star tree generation, even if _enableStarTreeIndex is not specified.
+      segmentGeneratorConfig.setEnableStarTreeIndex(starTreeIndexSpec);
+    } else if (_enableStarTreeIndex) {
+      segmentGeneratorConfig.setEnableStarTreeIndex(null);
```

Comment:
what is this for?

Reply:
So one can specify -enableStarTreeIndex without the config file.

Reply:
sorry if i'm misunderstanding, but if you're enabling startree, i assume you need some sort of config?